urld44fdf4fee_nadilsons_reflection_dsl_tgz-pJ8-br_com_bit_ideias_reflection_SimplePerformanceTestJ8/src/br/com/bit/ideias/reflection/core/extrator/ExtractorField.java

```diff
@@ -0,0 +1,125 @@
+package br.com.bit.ideias.reflection.core.extrator;
+
+import java.lang.reflect.Field;
+
+import br.com.bit.ideias.reflection.exceptions.FieldNotExistsException;
+import br.com.bit.ideias.reflection.exceptions.FieldPrivateException;
+import br.com.bit.ideias.reflection.exceptions.InvalidParameterException;
+import br.com.bit.ideias.reflection.exceptions.MethodNotExistsException;
+import br.com.bit.ideias.reflection.exceptions.StaticFieldNotExistsException;
+import br.com.bit.ideias.reflection.exceptions.StaticMethodNotExistsException;
+
+/**
+ * @author Nadilson Oliveira da Silva
+ * @date 19/02/2009
+ * 
+ */
+public class ExtractorField {
+
+	private final Extractor extractor;
+
+	private Field field;
+
+	private boolean directAccess;
+
+	ExtractorField(final Extractor extractor, final String fieldName) {
+		this.extractor = extractor;
+
+		final Class<?> targetClass = extractor.getTargetClass();
+		Class<?> actualClass = targetClass;
+
+		do {
+			try {
+				this.field = actualClass.getDeclaredField(fieldName);
+			} catch (final NoSuchFieldException e) {
+				// Nothing to do 
+			}
+			
+			if (this.field != null)
+				break;
+
+			actualClass = actualClass.getSuperclass();
+		} while (actualClass != null);
+
+		if (field == null)
+			throw new FieldNotExistsException(String.format("Field %s not exists in %s", fieldName, targetClass.getName()));
+	}
+
+	ExtractorField(Extractor extractor, Field field) {
+		this.extractor = extractor;
+		this.field = field;
+	}
+
+	public Object invoke(final boolean accessPrivateMembers, final Object... params) {
+		if (params.length > 1)
+			throw new InvalidParameterException(String.format("Número excessivo de parametros [%s] para o metodo setter", params.length));
+
+		try {
+			if(directAccess)
+				return invokeField(accessPrivateMembers, params);
+
+			return invokeMethod(accessPrivateMembers, params);
+		} catch (final IllegalAccessException e) {
+			throw new FieldPrivateException(e);
+		} catch (MethodNotExistsException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 71.